buckets/redis/bucket.go

```diff
@@ -114,11 +114,12 @@ func (b *redisBucket) Take(requested int64, maxWaitTime time.Duration) (time.Dur
 			waitTime = time.Nanosecond * time.Duration(waitTimeNanos)
 			keepTrying = false
 		default:
+			// Always close connections on errors to prevent results leaking.
+			b.factory.client.Close()
```

Comment:
It can, but I don't know if there's anything we can do about it. I can add a log statement.